### ISSUE TITLE
Invalid suffix of raw dataset when preprocessing without language

### ISSUE THREAD

**Issue by prihoda** [pytorch/fairseq#1426]
When preprocessing using `--dataset-impl raw` and no source and target languages are specified, the datasets are stored under `train.None-None` due to this line:

https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/preprocess.py#L218

Is this expected behavior or can we remove this suffix?

**Comment by myleott** [comment]
Yes it's expected behavior, since we generally use the `<split>.<srclang>-<tgtlang>` filename format. We could generalize it, but that would require a bit of extra logic in a few places. If you're passionate about fixing this, a PR is welcome :)

**Comment by prihoda** [comment]
And can you clarify why this suffix is used only for the `raw` dataset implementation?
The other implementations are already using this function to skip adding this suffix if the language is not specified:
https://github.com/pytorch/fairseq/blob/5349052aae4ec1350822c894fbb6be350dff61a0/preprocess.py#L322-L330
